src/core/deviceConnection/index.ts

```diff
@@ -449,14 +454,23 @@ export class DeviceConnection
       throw new Error('Only v3 packets are supported');
     }
 
-    const resp = await operations.sendAbort({
-      connection: this,
-      version,
-      sequenceNumber: params.sequenceNumber,
-      maxTries: params.maxTries
-    });
-
-    return resp;
+    const status = await this.getStatus({ maxTries: 2 });
+    if (
+      status.deviceIdleState === DeviceIdleState.USB &&
+      !status.abortDisabled
+    ) {
+      const resp = await operations.sendAbort({
+        connection: this,
+        version,
+        sequenceNumber: params.sequenceNumber,
+        maxTries: params.maxTries
+      });
+
+      return resp;
+    } else {
+      logger.info("Abort rejected based on device's current status ");
+      logger.info(status);
+    }
```

Comment:
Can you explore the possibility of storing the `abortDisabled` variable in the device connection itself, and use it when required?

So whenever we get a new status from device, we update the `abortDisabled` variable and use it in sendAbort function.